### PR TITLE
Apis

### PR DESCRIPTION
**Pull request by ShelOrock**
3 main API routes - Users, Orders, and Products.

Users can also load/create/edit/delete associated Orders.
Orders can also load/create/edit/delete associated Carts

Authentication Criteria: None. We can't test without a seeded database. But please check for typos and logic errors.

### CODE REVIEW

File: server/api/orders.js [hunk 1]
```diff
@@ -0,0 +1,84 @@
+const router = require("express").Router();
+
+// const { Order, User } = require("../db/index.js");
+
+//Finds and servers all orders
+router.get('/', (req, res, next) => {
+	Order.findAll()
+	.then(orders => res.send(orders))
```
Comment (nries1) [inline server/api/orders.js]: can you put a status 200 on this?

File: server/api/orders.js [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+//Eager loads associated user.
+router.get('/:id', (req, res, next) => {
+	Order.findByPk(req.params.id, {
+		include: [
+			{
+				model: User,
+			},
+		],
+	})
+	.then(order => res.send(order))
```
Comment (nries1) [inline server/api/orders.js]: status 200?

File: server/api/orders.js [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+//No need to update a userId
+router.put('/:id', (req, res, next) => {
+	const {
+		shippingAddress,
+		orderCost,
+	} = req.body
+
+	Order.findByPk(req.params.id)
+	.then(order => order.update({
+		shippingAddress: shippingAddress || user.shippingAddress,
```
Comment (nries1) [inline server/api/orders.js]: should be order.shippingAddress? and order.orderCost?

File: server/api/orders.js [hunk 4]
```diff
@@ -0,0 +1,84 @@
...
+		orderCost: orderCost || user.orderCost,
+	}))
+	.then(() => res.status(202))
+	.catch(e => {
+		res.status(304);
+		next(e);
+	})
+})
+
```
Comment (nries1) [inline server/api/orders.js:102]: It looks like all errors get passed down the middleware chain. Is there anything to catch them at the end?

File: server/api/products.js [hunk 5]
```diff
@@ -0,0 +1,78 @@
+const router = require("express").Router();
+
+// const { Product } = require("../db/index.js");
+
+
+router.get('/', (req, res, next) => {
+	Product.findAndCountAll()
+	.then(products => res.send(products))
```
Comment (nries1) [inline server/api/products.js]: status 200?

File: server/api/products.js [hunk 6]
```diff
@@ -0,0 +1,78 @@
...
+	const {
+		productName,
+		productDescription,
+		unitPrice,
+		inventory,
+	} = req.body
+
+	Product.create({
+		productName,
+		productDescription: productDescription || null,
```
Comment (nries1) [inline server/api/products.js]: Cool. Just noting that client needs to check if description is null b4 doing any string manipulation.

File: server/api/users.js [hunk 7]
```diff
@@ -0,0 +1,130 @@
+const router = require("express").Router();
+const paginate = require("./utils");
+
+// const { User, Order } = require('../db/index.js');
+
+//Finds, counts and serves all users
+router.get('/', paginate(User), (req, res, next) => {
+	res.send(paginate(res.paginate))
```
Comment (nries1) [inline server/api/users.js]: cool! excited to see this in action

File: server/api/users.js [hunk 8]
```diff
@@ -0,0 +1,225 @@
...
+		userType,
+		phone: phone || null,
+		shippingAddress: shippingAddress || null,
+		shippingCity: shippingCity || null,
+		shippingState: shippingState || null,
+		shippingZip: shippingZip || null,
+		billingAddress: billingAddress || null,
+		billingCity: billingCity || null,
+		billingState: billingState || null,
+		billingZip: billingZip || null,
```
Comment (raymondng1) [inline server/api/users.js:59]: When the new User row is created and added to the database will there also be a : 

'key(**loggedIn**)' default to 'false' column to be created as well?

**Review by raymondng1 — COMMENTED** [review]
Hey everything looks great for the routes! the logic on the cart route makes sense especially if a user closes and refreshes the page and wants to see their own cart items re-rendered again.

File: server/api/users.js [hunk 9]
```diff
@@ -0,0 +1,225 @@
...
+//Eager loads associated orders.
+router.get('/:userId/orders', (req, res, next) => {
+	User.findByPk(req.params.userId, {
+		include: [
+			{
+				model: Order,
+			},
+		],
+	})
+	.then(user => res.send(user))
```
Comment (raymondng1) [inline server/api/users.js]: maybe send status code 200 OK

File: server/api/users.js (hunk 8 above)
Comment (ShelOrock) [reply server/api/users.js:59]: If the database model gives loggedIn a default value of false this shouldn't be a problem.

**Review by raymondng1 — COMMENTED** [review]
The changes look great to me! I think Nick may want to double check as he made comments and reviewed this as well before merging to the Dev Branch.

**Review by nries1 — APPROVED** [review]
Looks good!

**Pull request merged**